js/web/lib/wasm/jsep/webgpu/ops/softmax.ts

```diff
@@ -52,36 +50,41 @@ const createSoftmaxProgramInfo = (input: TensorView, attributes: SoftmaxAttribut
 
     return name;
   };
-
+  const x = inputVariable('x', input.dataType, input.dims, components);
+  const output = outputVariable('result', input.dataType, input.dims, components);
+  const enableInputShapesUniforms = enableShapesUniforms(input.dims.length);
+  const programUniforms: ProgramUniform[] = [{type: 'uint32', data: packedCols}];
+  if (enableInputShapesUniforms) {
+    programUniforms.push(...createTensorShapeVariables(input.dims), ...createTensorShapeVariables(input.dims));
+  }
+  const inputDependencies: ProgramInputTensorInfoDependency[] = [enableInputShapesUniforms ? 'rank' : 'dims'];
```

Comment:
It seems that you only need to depend on `type` instead of `rank` when `enableInputShapesUniforms` is true.

Reply:
Done